### PR TITLE
integration: add constant requestWaitTimeout.

### PR DESCRIPTION
**Pull request by hexfusion**
I have begun stepping through the tests and defining constants. Before I get too far along I would like to get some general input on my naming and thoughts on extending constants via cluster. I appreciate you taking a look.

/cc @xiang90 @gyuho 

ref: https://github.com/coreos/etcd/issues/8781

### CODE REVIEW

File: clientv3/integration/lease_test.go [hunk 1]
```diff
@@ -299,7 +299,7 @@ func TestLeaseGrantErrConnClosed(t *testing.T) {
 	}
 
 	select {
-	case <-time.After(3 * time.Second):
+	case <-time.After(leaseRequestTimeout):
```
Comment (xiang90) [inline clientv3/integration/lease_test.go]: probably rename this to requestWaitTimeout? also comment this const as `time duration to wait for a request to go through`

**Comment by xiang90** [comment]
let us change the timeouts to consts one per PR? it is easier for review and discuss. 

I think the `requestWaitTimeout ` would be a good start.

**Comment by hexfusion** [comment]
@xiang90 yeah sounds good, thanks.

File: clientv3/integration/network_partition_test.go [hunk 2]
```diff
@@ -234,7 +234,7 @@ func testBalancerUnderNetworkPartitionWatch(t *testing.T, isolateLeader bool) {
 	wch := watchCli.Watch(clientv3.WithRequireLeader(context.Background()), "foo", clientv3.WithCreatedNotify())
 	select {
 	case <-wch:
-	case <-time.After(3 * time.Second):
+	case <-time.After(maxLeaderChangeWait):
```
Comment (xiang90) [inline clientv3/integration/network_partition_test.go]: this should also be a `requestTimeout` (the one we renamed from leaseRequestTimeout). basically we send a watch request to etcd and wait for the ack from the channel from the etcd server side.

**Comment by hexfusion** [comment]
Updated PR to focus on RequestWaitTimeout only.

**Comment by gyuho** [comment]
`clientv3/integration` should be importing `integration.RequestWaitTimeout`?

**Comment by hexfusion** [comment]
@gyuho your right import is not necessary, resolved.

File: clientv3/integration/watch_test.go [hunk 3]
```diff
@@ -760,7 +760,7 @@ func TestWatchWithRequireLeader(t *testing.T) {
 		if ok {
 			t.Fatalf("expected closed channel, got response %v", resp)
 		}
-	case <-time.After(3 * time.Second):
+	case <-time.After(requestWaitTimeout):
```
Comment (gyuho) [inline clientv3/integration/watch_test.go]: `integration.RequestWaitTimeout`?

File: integration/cluster.go [hunk 4]
```diff
@@ -58,10 +58,12 @@ import (
 )
 
 const (
-	tickDuration   = 10 * time.Millisecond
-	clusterName    = "etcd"
-	requestTimeout = 20 * time.Second
+	// time duration to wait for a request to go through or detect leader loss
```
Comment (fanminshi) [inline integration/cluster.go]: `// RequestWaitTimeout is the time duration...`?

Comment (fanminshi) [reply integration/cluster.go]: usually in go, we start the comment of an exported function/variable by state its name first.

Comment (hexfusion) [reply integration/cluster.go]: @fanminshi thank you!

File: clientv3/integration/watch_test.go (hunk 3 above)
Comment (hexfusion) [reply clientv3/integration/watch_test.go]: I see separate integration packages, will resolve.

File: clientv3/integration/dial_test.go [hunk 5]
```diff
@@ -57,7 +57,7 @@ func TestDialTLSExpired(t *testing.T) {
 	// expect remote errors "tls: bad certificate"
 	_, err = clientv3.New(clientv3.Config{
 		Endpoints:   []string{clus.Members[0].GRPCAddr()},
-		DialTimeout: 3 * time.Second,
+		DialTimeout: requestWaitTimeout,
```
Comment (xiang90) [inline clientv3/integration/dial_test.go]: this is dial timeout. why supply it with requestWaitTimeout?

File: clientv3/integration/network_partition_test.go [hunk 6]
```diff
@@ -110,7 +110,7 @@ func testBalancerUnderNetworkPartition(t *testing.T, op func(*clientv3.Client, c
 	// expect pin eps[0]
 	ccfg := clientv3.Config{
 		Endpoints:   []string{eps[0]},
-		DialTimeout: 3 * time.Second,
+		DialTimeout: requestWaitTimeout,
```
Comment (xiang90) [inline clientv3/integration/network_partition_test.go]: same question as above.

**Comment by xiang90** [comment]
lgtm. defer to @gyuho

**Comment by hexfusion** [comment]
thanks guys

**Comment by hexfusion** [comment]
@xiang90 I see a lot of time.Sleep(100 * time.Millisecond). I looked around and this pops out at me.

https://github.com/coreos/etcd/blob/6915a718b9ba366c53e12547cfe7c10c67f376bd/clientv3/client_test.go#L58-L60

So is this perhaps ClientDialDuration?

**Comment by hexfusion** [comment]
I also see this in raft tests as ticker duration. but then 

https://github.com/coreos/etcd/blob/6915a718b9ba366c53e12547cfe7c10c67f376bd/raft/node_test.go#L335

but then we define that as 10 Millisecond here.

https://github.com/coreos/etcd/blob/6915a718b9ba366c53e12547cfe7c10c67f376bd/integration/cluster.go#L63

**Comment by hexfusion** [comment]
If I run benchmark on a single node 10000 keys at 256 bytes the 99% percentile is about 10ms. 

```
$ ./benchmark --endpoints=http://127.0.0.1:2379 --target-leader --conns=1 --clients=1 put --key-size=8 --sequential-keys --total=10000 --val-size=256
INFO: 2018/01/04 07:12:38 ccResolverWrapper: sending new addresses to cc: [{localhost:2379 0  <nil>}]
 0 / 10000 B                                                                                                                          !   0.00%INFO: 2018/01/04 07:12:38 balancerWrapper: got update addr from Notify: [{localhost:2379 <nil>}]
INFO: 2018/01/04 07:12:38 balancerWrapper: got update addr from Notify: [{localhost:2379 <nil>}]
 10000 / 10000 Boooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo! 100.00% 22s

Summary:
  Total:	22.6218 secs.
  Slowest:	0.0401 secs.
  Fastest:	0.0003 secs.
  Average:	0.0022 secs.
  Stddev:	0.0019 secs.
  Requests/sec:	442.0523

Response time histogram:
  0.0003 [1]	|
  0.0043 [8890]	|∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎∎
  0.0083 [903]	|∎∎∎∎
  0.0123 [149]	|
  0.0163 [34]	|
  0.0202 [16]	|
  0.0242 [3]	|
  0.0282 [2]	|
  0.0322 [0]	|
  0.0361 [0]	|
  0.0401 [2]	|

Latency distribution:
  10% in 0.0015 secs.
  25% in 0.0016 secs.
  50% in 0.0017 secs.
  75% in 0.0018 secs.
  90% in 0.0045 secs.
  95% in 0.0047 secs.
  99% in 0.0100 secs.
  99.9% in 0.0186 secs.
```

**Comment by hexfusion** [comment]
moving this to new PR

**Pull request merged**